### PR TITLE
fix: alphabetical order

### LINKED ISSUE #19: run testsuites in alphabetical order with parallel = 1

### PR DESCRIPTION
**Pull request by yesnault**
close #19

Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

**Comment by fsamin** [comment]
I think that the process function is too big and the process file is always impacted and merges are painful.

I suggest to externalize the progress bar stuff, details level stuff, files reading, and failure computing is separate function and files.

**Comment by yesnault** [comment]
yep, doing

**Comment by yesnault** [comment]
refactor done

File: process.go [hunk 1]
```diff
@@ -1,329 +1,92 @@
...
-	log "github.com/Sirupsen/logrus"
-	"github.com/fsamin/go-dump"
 	"gopkg.in/cheggaaa/pb.v1"
-	"gopkg.in/yaml.v2"
 )
 
-var aliases map[string]string
-var bars map[string]*pb.ProgressBar
-var mutex = &sync.Mutex{}
+var (
```
Comment (fsamin) [inline process.go]: That's better !

I know it's boring and i'm a bit annoying but may you share all of those vars by function parameters, and avoid global vars ?

example: `readFiles` is using `pb`and `chanToRun`, it's a bit confusing without reading two other files.

Comment (yesnault) [inline process.go]: yep, done

**Comment by yesnault** [comment]
DO not merge, I have something to check on display results before

**Comment by yesnault** [comment]
refactor done bis

File: process.go [hunk 2]
```diff
@@ -1,339 +1,78 @@
...
-			continue
-		}
-
-		ts.Templater.Add(tc.Name, result)
-
-		log.Debugf("result:%+v", ts.Templater)
-
-		if h, ok := e.executor.(executorWithDefaultAssertions); ok {
-			isOK, errors, failures, systemout, systemerr = applyChecks(result, step, h.GetDefaultAssertions(), l)
+func computeStats(testsResult *Tests, chanEnd chan TestSuite, wg *sync.WaitGroup) {
```
Comment (fsamin) [inline process.go]: chanEnd <-chan TestSuit

File: process_files.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+			log.Fatalf("Error reading files on path:%s :%s", path, errg)
+		}
+		filesPath = append(filesPath, fp...)
+	}
+
+	sort.Strings(filesPath)
+	return filesPath
+}
+
+func readFiles(detailsLevel string, filesPath []string, chanToRun chan TestSuite) map[string]*pb.ProgressBar {
```
Comment (fsamin) [inline process_files.go]: chanToRun chan<- TestSuite

Comment (yesnault) [inline process_files.go]: fixed

**Pull request merged**